### DISCUSSION TITLE
Is it possible to access devices from external app using REST API

### DISCUSSION THREAD

**Discussion by Nasreddine** [microsoft/devicescript#635] [Q&A] [answered]
Hello,

I'm curious if there is a way to access the devices using a REST API? I've looked through the documentation, but I couldn't find any information on it.

Best regards,

**Comment by pelikhan** [comment]
We currently don't support running a web server from the device, so you would not be able to access them via REST.

You can do the other way around where the device opens a websocket to a known webserver and start communication with the server.

**Comment by Nasreddine** [comment]
I'm currently utilizing virtual devices. Although I'm not well-versed in TypeScript (TS), I attempted to include the Express package in order to expose sensor data through an API. Unfortunately, I encountered package errors in the process.
Please, what is it the best way to incorporate express or other package to support REST API ?

Thanks in advance.

**Comment by pelikhan** [comment, marked as answer]
Unfortunately, DeviceScript is rather different than node.js and express (or the http/https package won't work in that environment). If you need to do REST request from the device, you can use [fetch](https://microsoft.github.io/devicescript/developer/net#fetch)
